src/org/exist/xquery/lib/kwic.xql

```diff
@@ -17,34 +17,57 @@
     return<br/>
         kwic:get-summary($ancestor, ($ancestor//exist:match, $ancestor//*[@exist:matches])[1], $config)
     </pre>
-:)  
-module namespace kwic="http://exist-db.org/xquery/kwic";
+    
+    The configuration element used by some functions has the following structure:
+    
+    <config
+        width="character width"
+        table="yes|no"
+        link="URL to which the match is linked"/>
+:)
+xquery version "3.0";
+
+module namespace kwic = "http://exist-db.org/xquery/kwic";
+
+import module namespace util = "http://exist-db.org/xquery/util";
+
+declare namespace exist = "http://exist.sourceforge.net/NS/exist";
+
+declare variable $kwic:MODE_BEFORE := "before";
+declare variable $kwic:MODE_AFTER := "after";
+
+declare %private variable $kwic:CHARS_SUMMARY := 120;
+declare %private variable $kwic:CHARS_KWIC := 40;
 
-declare variable $kwic:CHARS_SUMMARY := 120;
-declare variable $kwic:CHARS_KWIC := 40;
 
 (:~
 	Like fn:substring, but takes a node argument. If the node is an element,
 	a new element is created with the same node-name as the old one and the
 	shortened text content.
 :)
-declare function kwic:substring($node as item(), $start as xs:int, $count as xs:int) as item()? {
+declare
+    %private
+function kwic:substring($node as item(), $start as xs:integer, $count as xs:integer) as item()? {
```

Comment:
would `node()` be more specific, since we're now returning either an `element()` or `text()`?